scripts/check_compliance.py

```diff
@@ -301,6 +289,18 @@ def check_top_menu_not_too_long(self, kconf):
 entries, then bump the 'max_top_items' variable in {}.
 """.format(max_top_items, n_top_items, __file__))
 
+    def check_no_undef_within_kconfig(self, kconf):
+        """
+        Checks that there are no references to undefined Kconfig symbols within
+        the Kconfig files
+        """
+        undef_ref_warnings = "\n\n\n".join(warning for warning in kconf.warnings
+                                           if "undefined symbol" in warning)
+
+        if undef_ref_warnings:
+            self.add_failure("Undefined Kconfig symbols:\n\n"
+                             + undef_ref_warnings)
+
```

Comment:
Nit: the commit message should be more explicit that the code hasn't changed at all, would save time.